packages/core/src/State.ts

```diff
@@ -245,6 +251,27 @@ export class State<
     return isInFinalState(this.configuration);
   }
 
+  /**
+   * The done data of the top-level finite state.
+   */
+  public get output(): any {
```

Comment:
This requires some type-related work in a followup OR or something. We shouldn't ever use `any` in the public API.

Reply:
Also, WDYT about not relying on getters? IMHO they are kinda obscure because when reading the code consuming this it's not apparent that there is some extra magic happening here under the hood. This is a lie to the user and I've been bitten by this a lot of times when debugging code - because when I see property access I just step over it as I expect it to be a simple operation.

The last time I checked getters were also associated with an engine perf penalty when compared to simple functions.

Reply:
I added a TODO to add an explicit type for `output`, which is meant to represent the final "done data" (output is a better name IMO, symmetrical with input) of the machine. This did not exist before because done data was simply sent up to the parent instead of being represented on the state.

Re: not relying on getters, I suppose these can be computed in the constructor 🤔